### DISCUSSION TITLE
Continuous flux through hooks and global initial hook var

### DISCUSSION THREAD

**Discussion by bilicio** [feathersjs/feathers#3390] [Q&A] [answered]
Hi guys, Im using a hook to do some things through a process however, if i use a before hook, the main process is never called. This is a normal behavior? This is the process im doing, i call the api like https://site.com/form, then i use a before hook like this:
```
export const generatePix = async (context) => {
    const postServerGeneratePix = async (data) => {
      return new Promise((resolve, reject) => {
        return axios
          .post(url, data, {
            headers: {
              accept: 'application/json',
              'content-type': 'application/json',
            }
          })
          .then((response) => {
            return resolve(response)
          })
          .catch((error) => {
            return reject(error.response.data)
          })
      })
    }

    const response = await postServerGeneratePix(context.data)
    context.result = response
    return context

}
```
The hook is executed normally, however im not getting back to the usual create api, this is not executed:
```
async create(data, params) {
    if (Array.isArray(data)) {
      return Promise.all(data.map((current) => this.create(current, params)))
    }

    return {
      id: 0,
      ...data
    }
  }
```
  I tried to use next() but im getting the message `next() is not a function`, even when i declare side by side with the context `(context, next)`.
  Another thing, how can i use the setup hook to keep the object of my initialized database to be available in every service? like, this is my hook:
  ```
setup: [
    async (context) => {
      const pb = new PocketBase('https://server.com/')
      context.params.pb = pb
    }
  ],
```
Im trying keep the database available everywhere.
Can you guys give me a light :)

**Comment by daffl** [comment]
What does your hook look like? How are you registering it?

`setup` is only called once when the application starts up. Usually you'd register a database connection via [app.set]https://feathersjs.com/api/application.html#set-name-value) and then pass it to the service in the constructor:

```ts
app.set('pb', new PocketBase('https://server.com/'))

app.use('myService', new MyService(app.get('pb'))
```

**Reply by bilicio** [reply]
Hi @daffl :)
I putted my hook on the corresponding service:
```app.service(cadastroPath).hooks({
    around: {
      all: [
        schemaHooks.resolveExternal(cadastroExternalResolver),
        schemaHooks.resolveResult(cadastroResolver)
      ]
    },
    before: {
      all: [
        schemaHooks.validateQuery(cadastroQueryValidator),
        schemaHooks.resolveQuery(cadastroQueryResolver)
      ],
      find: [],
      get: [],
      create: [
        schemaHooks.validateData(cadastroDataValidator),
        schemaHooks.resolveData(cadastroDataResolver),
        generatePix // Here is the hook
      ],
      patch: [
        schemaHooks.validateData(cadastroPatchValidator),
        schemaHooks.resolveData(cadastroPatchResolver)
      ],
      remove: []
    },
    after: {
      all: [],
      create: [
        async (context) => {
          await pb.collection('tickets').create(context.result)
          context.result = { customer: context.result.customer }
        }
      ]
    },
    error: {
      all: [errorHandler]
    }
  })
  ```
`  About the setup, I will try your aproach` 👍🏻

**Reply by daffl** [reply, marked as answer]
Yes but what is `generatePix` actually doing? If it sets `context.result` the actual service call _will_ be skipped.

**Reply by bilicio** [reply]
Oh! thats it? now i got it. But if i need to exchange information between the hook and the actual service i cant use `context.result`, in which object its better to send information? `context.?`

**Reply by daffl** [reply]
You can add anything you'd like to `context.params`.

**Reply by bilicio** [reply]
Worked david! thank you!
